shared-module/gamepad/GamePad.c

```diff
@@ -24,37 +24,34 @@
  * THE SOFTWARE.
  */
 
-#include <stdbool.h>
-
 #include "py/mpstate.h"
-#include "__init__.h"
-#include "GamePad.h"
-
-#include "shared-bindings/digitalio/Pull.h"
 #include "shared-bindings/digitalio/DigitalInOut.h"
-#include "shared-bindings/util.h"
-
+#include "shared-bindings/gamepad/GamePad.h"
 
-void gamepad_init(size_t n_pins, const mp_obj_t* pins) {
-    gamepad_obj_t* gamepad_singleton = MP_STATE_VM(gamepad_singleton);
+void common_hal_gamepad_gamepad_init(gamepad_obj_t *gamepad,
+                  const mp_obj_t pins[], size_t n_pins) {
     for (size_t i = 0; i < 8; ++i) {
-        gamepad_singleton->pins[i] = NULL;
+        gamepad->pins[i] = NULL;
     }
-    gamepad_singleton->pulls = 0;
+    gamepad->pulls = 0;
     for (size_t i = 0; i < n_pins; ++i) {
         digitalio_digitalinout_obj_t *pin = MP_OBJ_TO_PTR(pins[i]);
-        digitalio_direction_t direction = common_hal_digitalio_digitalinout_get_direction(pin);
-        if (direction != DIRECTION_INPUT) {
+        if (common_hal_digitalio_digitalinout_get_direction(pin) !=
+            DIRECTION_INPUT) {
             common_hal_digitalio_digitalinout_switch_to_input(pin, PULL_UP);
         }
         digitalio_pull_t pull = common_hal_digitalio_digitalinout_get_pull(pin);
         if (pull == PULL_NONE) {
             common_hal_digitalio_digitalinout_set_pull(pin, PULL_UP);
         }
         if (pull != PULL_DOWN) {
```

Comment:
@deshipu - It looks here that the DigitalInOut objects are made to be inputs with PULL_UP if they are currently outputs. Also if they are already inputs with PULL_NONE, they will be made to be PULL_UP. But if they are inputs with PULL_DOWN, they won't be changed. Is that your goal?

In `shared-bindings/gamepad/GamePad.c`, the provided doc says they are always pulled up. Could you change the doc to match the impl (unless I am misunderstanding)?
```
//|     The ``b1``-``b8`` parameters are ``DigitalInOut`` objects, which
//|     immediately get switched to input with a pull-up, and then scanned
//|     regularly for button presses. The order is the same as the order of
//|     bits returned by the ``get_pressed`` function. You can re-initialize
//|     it with different keys, then the new object will replace the previous
//|     one.
```

Reply:
I'm not sure how to add commits to this pull request, now that @tannewt added his. I can fix it in a separate PR?

Reply:
Fixed in #1808

Reply:
The new PR is fine.